src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -609,7 +609,15 @@ public uint256 LastReceivedBlockHash()
                     .OrderBy(o => o.LastBlockSyncedHeight)
                     .FirstOrDefault()?.LastBlockSyncedHash;
 
-                Guard.Assert(lastBlockSyncedHash != null);
+                // If details about the last block synced are not present in the wallet,
+                // find out which is the oldest wallet and set the last block synced to be the one at this date.
+                if (lastBlockSyncedHash == null)
+                {
+                    this.logger.LogWarning("There were no details about the last block synced in the wallets.");
+                    DateTimeOffset earliestWalletDate = this.Wallets.Min(c => c.CreationTime);
+                    this.UpdateWhenChainDownloaded(this.Wallets, earliestWalletDate.DateTime);
```

Comment:
What happens if there are no wallets? Or the list is empty?

Reply:
a bit above this code, we check if there are no wallets, in which case we return chain.tip.

Reply:
Ahhh sorry yes!